CMake issue: Cannot specify link libraries for target "pugixml" which is not built by this project

Hi,

I have this CMake issue with latest pugixml (9079552) with CMake 3.10.2 that is the latest available in Ubuntu 18.04:

> CMake Error at /usr/local/lib/cmake/pugixml/pugixml-config.cmake:17 (target_link_libraries):
   Cannot specify link libraries for target "pugixml" which is not built by
   this project.
 Call Stack (most recent call first):
   CMakeLists.txt:27 (find_package)

At this line, there is:

```
find_package(pugixml REQUIRED)
```

There is no issue with `cmake-3.16.5.tar.gz`. So this is probably a CMake issue.
~~There is also no issue with CMake 3.10.2 with this change (though I don't know if it is correct or not in CMake world):~~

```
diff --git a/scripts/pugixml-config.cmake.in b/scripts/pugixml-config.cmake.in
index 58be639..162884e 100644
--- a/scripts/pugixml-config.cmake.in
+++ b/scripts/pugixml-config.cmake.in
@@ -6,6 +6,6 @@ include("${CMAKE_CURRENT_LIST_DIR}/pugixml-targets.cmake")
 # version or not requesting one at all), provide the old imported target name
 # for compatibility.
 if (NOT DEFINED PACKAGE_FIND_VERSION OR PACKAGE_FIND_VERSION VERSION_LESS "1.11")
-  add_library(pugixml INTERFACE IMPORTED)
+  add_library(pugixml INTERFACE)
   target_link_libraries(pugixml INTERFACE pugixml::pugixml)
 endif ()
```

Edit:
it does not work with multiple projects:

```
CMake Error at /usr/local/lib/cmake/pugixml/pugixml-config.cmake:16 (add_library):
  add_library cannot create target "pugixml" because another target with the
  same name already exists.  The existing target is an interface library
  created in source directory
```

I have two questions:

- ~~is it possible to point me to the commit that introduces these CMake changes? Because before it works flawlessly, and I would like to checkout this specific commit state~~ Looks like v1.11 works for me.
- is it possible to keep compatibility with CMake 3.10.2/Ubuntu 18.04?

I can reproduce this on CMake 3.10 on Linux with v1.11.2

```
CMake Error at /mnt/c/work/pugixml-build/pugixml-config.cmake:17 (target_link_libraries):
  Cannot specify link libraries for target "pugixml" which is not built by
  this project.
Call Stack (most recent call first):
  CMakeLists.txt:4 (find_package)
```

v1.11 works only if you specify pugixml::pugixml as a dependency (as expected)
v1.11.1 fails with the following message (which is why v1.11.2 was created in the first place):

```
CMake Error at /mnt/c/work/pugixml-build/pugixml-config.cmake:16 (add_library):
  add_library cannot create ALIAS target "pugixml" because target
  "pugixml::pugixml" is IMPORTED.
```

With CMake 3.10.2, `v1.11` is the last version that works with my projects.

The patch I mentioned does not work. I have initially tested it on a simple project, but the patch does not work with my (rather) complicated projects (multiple submodules and multiple inclusions of pugixml). The error with this patch is:

```
CMake Error at /usr/local/lib/cmake/pugixml/pugixml-config.cmake:16 (add_library):
  add_library cannot create target "pugixml" because another target with the
  same name already exists.  The existing target is an interface library
  created in source directory
```

---

I think it would be worth to add in the CI simple CMake tests. Since I am working on Ubuntu, CMake tests for the recent LTS versions (18.04, 20.04) would be great.

--- a/scripts/pugixml-config.cmake.in
+++ b/scripts/pugixml-config.cmake.in
@@ -7,5 +7,6 @@ include("${CMAKE_CURRENT_LIST_DIR}/pugixml-targets.cmake")
 # for compatibility.
 if (NOT DEFINED PACKAGE_FIND_VERSION OR PACKAGE_FIND_VERSION VERSION_LESS "1.11")
   add_library(pugixml INTERFACE IMPORTED)
-  target_link_libraries(pugixml INTERFACE pugixml::pugixml)
+  # Equivalent to target_link_libraries INTERFACE, but compatible with CMake 3.10
+  set_target_properties(pugixml PROPERTIES INTERFACE_LINK_LIBRARIES pugixml::pugixml)
 endif ()

